File: playify/android/src/main/kotlin/com/kaya/playify/playify/Classes/Album.kt
package com.kaya.playify.playify.Classes

class Album {
    constructor(title: String? = null, songs: List<Song>? = null, artistName: String? = null,
                albumTrackCount: Int? = null, coverArt: Array<Int>? = null, discCount: Int? = null) {
        this.title = title
        this.songs = songs
        this.artistName = artistName
        this.albumTrackCount = albumTrackCount
        this.coverArt = coverArt
        this.discCount = discCount
    }

    ///The title of the album.
    var title: String?

    ///The songs of the album.
    var songs: List<Song>?

    ///The name of the artist of the album.
    var artistName: String?

    ///The total tracks that the album has.
    ///
    ///This may not be equal to the total songs of the album that a user has on
    ///their device, since a song in an album may not exist on the device.
    var albumTrackCount: Int?

    ///The cover art as a `UInt8List`. This can be used with `Image.memory()` to
    ///convert to an image and display in the UI.
    var coverArt: Array<Int>?

    ///The total disc number of the album.
    var discCount: Int?

    override fun toString(): String {
        return "Album(title=$title, songs=$songs, artistName=$artistName, albumTrackCount=$albumTrackCount, coverArt=${coverArt?.contentToString()}, discCount=$discCount)"
    }
}

File: playify/lib/src/class/album/album.dart
import 'package:playify/src/class/song/song.dart';
import 'dart:typed_data';

class Album {
  Album(
      {required this.title,
      required this.songs,
      required this.albumTrackCount,
      required this.artistName,
      required this.coverArt,
      required this.discCount});

  ///The title of the album.
  String title;

  ///The songs of the album.
  List<Song> songs;

  ///The name of the artist of the album.
  String artistName;

  ///The total tracks that the album has.
  ///
  ///This may not be equal to the total songs of the album that a user has on
  ///their device, since a song in an album may not exist on the device.
  int albumTrackCount;

  ///The cover art as a `UInt8List`. This can be used with `Image.memory()` to
  ///convert to an image and display in the UI.
  Uint8List? coverArt;

  ///The total disc number of the album.
  int discCount;

  @override
  String toString() {
    return 'Title: ' +
        title +
        ' has ' +
        songs.length.toString() +
        ' songs and artist is ' +
        artistName +
        '\n';
  }
}
